specifyweb/frontend/js_src/lib/components/Molecules/Dialog.tsx

```diff
@@ -364,33 +361,42 @@ export function Dialog({
     );
   }, [showIcon, defaultIcon, buttons, buttonContainer]);
 
-  const overlayElement: Props['overlayElement'] = React.useCallback(
-    (
-      props: React.ComponentPropsWithRef<'div'>,
-      contentElement: React.ReactElement
-    ) => (
-      <div
-        {...props}
-        onMouseDown={
-          closeOnOutsideClick
-            ? (event): void => {
-                // Outside click detection
-                if (
-                  modal &&
-                  typeof handleClose === 'function' &&
-                  event.target === event.currentTarget
-                ) {
-                  event.preventDefault();
-                  handleClose();
-                } else props?.onMouseDown?.(event);
-              }
-            : undefined
-        }
-      >
-        {contentElement}
-      </div>
-    ),
-    [modal, handleClose, closeOnOutsideClick]
+  const allowCloseRef = React.useRef<boolean>(false);
+  React.useEffect(() => {
+    const timer = setTimeout(() => {
+      allowCloseRef.current = true;
+    }, 300);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, []);
+
+  const overlayElement: Props['overlayElement'] = (
```

Comment:
this function was using React.useCallback before
though, according to the library's source code, they don't care if this prop is memoized - https://github.com/reactjs/react-modal/blob/a275399059cc37aa02cac7e9385512b02ff5cf15/src/components/ModalPortal.js#L414
so you did good by removing a needless useCallback!